### PR TITLE
chore: [PAR-4171] add linting PR check via github action for magento module repo

### PR DESCRIPTION
**Pull request by milesmcleod**
* new pr-checks.yml lints js, php, and xml
  * currently, this lints just the files detected by running a git diff between the most recent commit on the current branch and the target branch. we should change this to lint the entire file system once we fix the formatting issues with the files that are already in the repo.
* yarn.lock is updated to use public npm registry. this means we don't need any AWS creds to run `yarn install` in GHA. _please let me know if this is the wrong approach._

### CODE REVIEW

File: .github/CODEOWNERS
Comment (billhefty) [inline .github/CODEOWNERS:1]: Should we go ahead and add our team here?

**Review by billhefty — DISMISSED** [review]

File: .github/CODEOWNERS
Comment (milesmcleod) [reply .github/CODEOWNERS:1]: our team is actually already here, this is just moving it into .github/. just doesn't show in the diff

Comment (billhefty) [reply .github/CODEOWNERS:1]: Oh, weird it wasn't showing any content in Github lol

**Review by billhefty — APPROVED** [review]
👍

**Review by jay-faust-extend — APPROVED** [review]

**Pull request merged**